### ISSUE TITLE
SDL2: display.set_mode() defaults to minimum width on windows

### ISSUE THREAD

**Issue by charlesej** [pygame/pygame#993]
The display width defaults to a minimum value when setting the mode for pygame SDL2 builds. This does not occur for pygame SDL1.2 builds.

Code example to show the width being defaulted to a minimum value.
```python
import pygame

pygame.display.init()
surface = pygame.display.set_mode((50, 50))
print('surface size =', surface.get_size())
pygame.display.quit()
```

Output from the above code. The requested width was 50, but it has been set to 120.
```
pygame 2.0.0.dev0 (SDL 2.0.9, python 3.7.2)
Hello from the pygame community. https://www.pygame.org/contribute.html
surface size = (120, 50)
```

System details:
- os: windows 10 (64bit)
- python: 3.7.2 (64bit)
- pygame: 2.0.0.dev0 (SDL: 2.0.9) at 0f051ad9423350a215a43ac72f1c98581f22579e

**Related Docs**: https://www.pygame.org/docs/ref/display.html#pygame.display.set_mode

**Comment by charlesej** [comment]
If `display.set_mode` is called twice, the width is set correctly the second time.

```python
import pygame

pygame.display.init()
surface = pygame.display.set_mode((50, 50))
print('surface size =', surface.get_size())

surface = pygame.display.set_mode((50, 50))
print('surface size =', surface.get_size())
pygame.display.quit()
```

Output from the above code.
```
pygame 2.0.0.dev0 (SDL 2.0.9, python 3.7.2)
Hello from the pygame community. https://www.pygame.org/contribute.html
surface size = (120, 50)
surface size = (50, 50)
```

**Comment by husano896** [comment]
This seems to be a SDL2 problem.

using the latest version (SDL v2.0.9)

    window = SDL_CreateWindow("Test", 0, 0, 20, 20, 0);
    int check_w, check_h;
    SDL_GetWindowSize(window, &check_w, &check_h);
    //Shows 120,20
    std::cout << check_w << "," << check_h;

**Comment by robertpfeiffer** [comment]
I'm not seeing this bug on Ubuntu with SDL 2.0.8

**Comment by illume** [comment]
I guess this is a windows bug with SDL 2.0.9. @charlesej is using windows, and I think @husano896 is using windows too.

Mac, SDL 2.0.10 works.
```
('surface size =', (50, 50))
```

So we should make a bug report on https://bugzilla.libsdl.org/

**Comment by robertpfeiffer** [comment]
I'll see if we already have a test for this, and if not write a test and close the issue

**Comment by nthykier** [comment]
@robertpfeiffer Did you find if we had a test for this, so we can close the issue? :)

**Comment by MyreMylar** [comment]
**To Do to close this:**

- [ ] Test the code above on windows in 2.0.0.dev8
- [ ] Write a unit test for sub 120 pixel width surfaces.

**Comment by MyreMylar** [comment]
This still occurs in 2.0.0.dev8 with SDL 2.0.12.

> @robertpfeiffer Did you find if we had a test for this, so we can close the issue? :)

Looking at the set_mode tests there is already a test that covers this use case that is being skipped on SDL2:

    # This decorator can be removed (or test changed) when issues #991 and #993
    # are resolved.
    @unittest.skipIf(SDL2, "SDL2 issues")
    def test_get_surface(self):
        """Ensures get_surface gets the current display surface."""
        lengths = (1, 5, 100)

        for expected_size in ((w, h) for w in lengths for h in lengths):
            for expected_depth in (8, 16, 24, 32):
                expected_surface = display.set_mode(expected_size, 0, expected_depth)

                surface = pygame.display.get_surface()

                self.assertEqual(surface, expected_surface)
                self.assertIsInstance(surface, pygame.Surface)
                self.assertEqual(surface.get_size(), expected_size)
                self.assertEqual(surface.get_bitsize(), expected_depth)

Perhaps we could spilt this into two tests, one covering the size which is skipped on windows and one covering the depth which is skipped on all platforms.

I raised a bug on the SDL Bugzilla here:

https://bugzilla.libsdl.org/show_bug.cgi?id=5144

**Comment by ankith26** [comment]
I'm testing on windows 11.

I can reproduce this issue in pygame 2.1.3dev6 (SDL 2.0.22) but this issue is fixed in SDL 2.24.0
